x/mongo/driver/operation/count.go

```diff
@@ -102,16 +126,56 @@ func (c *Count) Execute(ctx context.Context) error {
 		ServerAPI:         c.serverAPI,
 	}.Execute(ctx, nil)
 
+	// Swallow error if NamespaceNotFound(26) is returned from aggregate on non-existent namespace
+	if err != nil {
+		dErr, ok := err.(driver.Error)
+		if ok && dErr.Code == 26 {
+			err = nil
+		}
+	}
+	return err
 }
 
 func (c *Count) command(dst []byte, desc description.SelectedServer) ([]byte, error) {
-	dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+	// If wire version < 12 (4.9.0), use count command
+	if desc.WireVersion.Max < 12 {
+		dst = bsoncore.AppendStringElement(dst, "count", c.collection)
+		if c.maxTimeMS != nil {
```

Comment:
`maxTimeMS` applies for both so can we refactor this method to have this structure:

```
switch {
case desc.WireVersion.Max < 12:
    // existing code for count command but without maxTimeMS conditional
default:
    // existing code for aggregate but without maxTimeMS conditional
}

if c.maxTimeMS != nil {
    // add field
}
return dst, nil
```

Reply:
Seems cleaner! Done.